flang/test/Driver/lto-flags.f90

```diff
@@ -10,7 +10,8 @@
 ! RUN: %flang -### -S -flto=jobserver %s 2>&1 | FileCheck %s --check-prefix=FULL-LTO
 
 ! Also check linker plugin opt for Thin LTO
-! RUN: %flang -### -flto=thin %s 2>&1 | FileCheck %s --check-prefix=THIN-LTO
+! RUN: %flang -### -flto=thin %s 2>&1 | FileCheck %s --check-prefixes=%if \
+! RUN:        system-darwin %{THIN-LTO%} %else %{THIN-LTO,THIN-LTO-NOT-DARWIN%}
```

Comment:
[nit] You could rename:
*  `THIN-LOT` as `THIN-LTO-ALL` (it applies to ALL targets), and 
* `THIN-LOT-NOT-DARWIN` as `THIN-LTO-LINKER-PLUGIN` (this is a linker plugin "thingy" that you just wouldn't test for on Darwin as it's not supported).

I'm only lightly concerned that using "NOT" in a prefix might be confused with "CHECK-NOT" (which means something else).

Reply:
Changed in last commit.